src/js/components/InfiniteScroll/InfiniteScroll.js

```diff
@@ -177,7 +177,7 @@ const InfiniteScroll = ({
       }
     }, 100);
     return () => clearTimeout(timer);
-  }, [renderPageBounds, show, step]);
+  }, [renderPageBounds, show, step, scrollShow]);
```

Comment:
Why is this PR marked as draft?

Reply:
Marked as draft per this discussion: https://github.com/grommet/grommet/pull/5309#issuecomment-857232042. There was a concern that including dependency could result in an infinite loop, so want to hold off merging until edge cases are investigated.

Reply:
I believe the if statement on Line 160 accounts for and addresses the concern. I also have not been able to reproduce any cases invoking the infinite loop.

Regardless, if the decision is made to not include `scrollShow` as a dependency, then a comment should be added so that the rationale is documented.